src/VisualStudio/Core/Impl/Options/AbstractOptionPage.cs

```diff
@@ -81,19 +89,36 @@ public override void LoadSettingsFromStorage()
             // they may have been changed programmatically. Therefore, we'll set a flag so we load
             // next time
 
-            s_needsLoadOnNextActivate = pageControl != null;
+            // When pageControl is null we know that Activation has not happened for this page.
+            // We only need to update the OptionStore after a cancel or close click (2).
+            s_needsToUpdateOptionStore = pageControl != null;
+
+            // For pages that don't use option bindings we need to load settings when it is
+            // activated next.
+            _needsLoadOnNextActivate = true;
         }
 
         public override void SaveSettingsToStorage()
         {
             EnsureOptionPageCreated();
 
+            // Allow page controls to perist their settings to the option store before updating the
+            // option service.
+            pageControl.OnSave();
+
             // Save the changes that were accumulated in the option store.
-            s_optionService.SetOptions(s_optionStore.GetOptions());
+            var oldOptions = s_optionService.GetOptions();
+            var newOptions = s_optionStore.GetOptions();
+            s_optionService.SetOptions(newOptions);
+            OptionLogger.Log(oldOptions, newOptions);
```

Comment:
As pointed out by David offline, we had been logging option changes on SaveSettings before the earlier PR moved logging into the OptionStore. This reverts the behavior back.